packages/core/src/types.ts

```diff
@@ -196,7 +196,11 @@ export type Configuration = {
 
 export type Locale = string
 export type i18nOptions = Record<Locale, i18n>
-export type i18n = typeof en
+// allows partial i18n override in TypeScript:
+type RecursivePartial<T> = { //https://stackoverflow.com/a/47914631
+  [P in keyof T]?: RecursivePartial<T[P]>;
```

Comment:
```suggestion
type RecursivePartial<T> = { 
  // https://stackoverflow.com/a/47914631
  [P in keyof T]?: RecursivePartial<T[P]>;
```
Not sure if this link is even needed as it just points to stackoverflow, some explaining text would be more helpful